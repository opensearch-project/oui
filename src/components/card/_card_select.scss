/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

.ouiCardSelect {
  // Option select button that expands to sides and bottom
  @include ouiCardBottomNodePosition;

  font-weight: $ouiFontWeightBold;

  // Create button modifiers based upon the map.
  @each $name, $color in $ouiCardSelectButtonBackgrounds {
    &--#{$name}:enabled {
      background-color: $color;

      // Custom success text color since it doesn't exist on OuiButtonEmpty
      @if  $name == 'success' {
        color: makeHighContrastColor($ouiColorSuccess, $color);
      }
    }
  }

  &:disabled {
    background-color: $ouiPageBackgroundColor;
  }
}
