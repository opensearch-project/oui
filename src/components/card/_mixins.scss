/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

@mixin ouiCardBottomNodePosition {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: $ouiCardBottomNodeHeight !important;
  overflow: hidden;

  // Subtract 1px from border radius since it sits inside a border-radius
  border-bottom-left-radius: $ouiBorderRadius - 1px;
  border-bottom-right-radius: $ouiBorderRadius - 1px;
}


/* OUI -> EUI Aliases */
@mixin euiCardBottomNodePosition { @include ouiCardBottomNodePosition; }

/* End of Aliases */
