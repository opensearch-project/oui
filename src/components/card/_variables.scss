/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

$ouiCardSpacing: map-get($ouiPanelPaddingModifiers, 'paddingMedium');
$ouiCardBottomNodeHeight: $ouiSizeXXL;
$ouiCardSelectButtonBorders: (
  text: $ouiColorSuccess, // Default for selected
  primary: $ouiColorPrimary,
  success: $ouiColorSuccess,
  danger: $ouiColorDanger,
  ghost: $ouiColorDarkShade,
);
$ouiCardSelectButtonBackgrounds: (
  text: $ouiColorLightestShade, // Default for unselected
  primary: tintOrShade($ouiColorPrimary, 90%, 70%),
  success: tintOrShade($ouiColorSuccess, 90%, 70%),
  danger: tintOrShade($ouiColorDanger, 90%, 70%),
  ghost: $ouiColorDarkShade,
);
$ouiCardPaddingModifiers: (
  'paddingNone': 0,
  'paddingSmall': $ouiSizeS,
  'paddingMedium': $ouiSize,
  'paddingLarge': $ouiSizeL
) !default;


/* OUI -> EUI Aliases */
$euiCardSpacing: $ouiCardSpacing;
$euiCardBottomNodeHeight: $ouiCardBottomNodeHeight;
$euiCardSelectButtonBorders: $ouiCardSelectButtonBorders;
$euiCardSelectButtonBackgrounds: $ouiCardSelectButtonBackgrounds;
$euiCardPaddingModifiers: $ouiCardPaddingModifiers;

/* End of Aliases */
