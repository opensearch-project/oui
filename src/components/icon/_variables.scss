/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

$ouiIconLoadingOpacity: .05;

// Modifier naming and colors.
$ouiIconColors: (
  accent: makeGraphicContrastColor($ouiColorAccent),
  danger: makeGraphicContrastColor($ouiColorDanger),
  ghost: $ouiColorGhost,
  primary: makeGraphicContrastColor($ouiColorPrimary),
  secondary: makeGraphicContrastColor($ouiColorSecondary),
  success: makeGraphicContrastColor($ouiColorSuccess),
  subdued: $ouiTextSubduedColor,
  text: $ouiTextColor,
  warning: makeGraphicContrastColor($ouiColorWarning),
  inherit: inherit
);
$ouiIconSizes: (
  small: $ouiSizeM,
  medium: $ouiSize,
  large: $ouiSizeL,
  xLarge: $ouiSizeXL,
  xxLarge: $ouiSizeXXL
);


/* OUI -> EUI Aliases */
$euiIconLoadingOpacity: $ouiIconLoadingOpacity;
$euiIconColors: $ouiIconColors;
$euiIconSizes: $ouiIconSizes;

/* End of Aliases */
