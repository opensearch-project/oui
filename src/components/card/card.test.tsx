/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { requiredProps } from '../../test';

import { OuiCard } from './card';

import { OuiIcon } from '../icon';
import { OuiI18n } from '../i18n';
import { COLORS, SIZES } from '../panel/panel';

describe('OuiCard', () => {
  test('is rendered', () => {
    const { container } = render(
      <OuiCard
        title="Card title"
        description="Card description"
        {...requiredProps}
      />
    );

    expect(container).toMatchSnapshot();
  });

  describe('props', () => {
    test('icon', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          icon={<OuiIcon className="myIconClass" type="apmApp" />}
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('a null icon', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          icon={null}
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('hasBorder', () => {
      const { container } = render(
        <OuiCard title="Card title" description="Card description" hasBorder />
      );

      expect(container).toMatchSnapshot();
    });

    test('horizontal', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          layout="horizontal"
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('image', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          image={
            <div>
              <img
                src="https://source.unsplash.com/400x200/?Nature"
                alt="Nature"
              />
            </div>
          }
        />
      );

      expect(container).toMatchSnapshot();
    });

    describe('href', () => {
      it('supports href as a link', () => {
        const { container } = render(
          <OuiCard title="Hoi" description="There" href="#" />
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick as a link', async () => {
        const handler = jest.fn();
        render(
          <OuiCard title="Hoi" description="There" href="#" onClick={handler} />
        );

        const user = userEvent.setup();
        const link = screen.getByRole('link');

        await act(async () => {
          await user.click(link);
        });

        expect(handler).toHaveBeenCalledTimes(1);
      });

      it('supports onClick as a button', async () => {
        const handler = jest.fn();
        render(<OuiCard title="Hoi" description="There" onClick={handler} />);

        const user = userEvent.setup();
        const button = screen.getByRole('button');

        await act(async () => {
          await user.click(button);
        });

        expect(handler).toHaveBeenCalledTimes(1);
      });
    });

    test('titleElement', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          titleElement="h4"
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('titleElement with nodes', () => {
      const { container } = render(
        <OuiCard
          title={
            <OuiI18n token="ouiCard.title" default="Card title" /> // eslint-disable-line
          }
          description="Card description"
          titleElement="h4"
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('titleSize', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          titleSize="xs"
        />
      );

      expect(container).toMatchSnapshot();
    });

    describe('accepts div props', () => {
      test('like style', () => {
        const { container } = render(
          <OuiCard
            title="Card title"
            description="Card description"
            style={{ minWidth: 0 }}
          />
        );

        expect(container).toMatchSnapshot();
      });
    });

    test('footer', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          footer={<span>Footer</span>}
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('children', () => {
      const { container } = render(<OuiCard title="Card title">Child</OuiCard>);

      expect(container).toMatchSnapshot();
    });

    test('children with description', () => {
      const { container } = render(
        <OuiCard title="Card title" description="Card description">
          Child
        </OuiCard>
      );

      expect(container).toMatchSnapshot();
    });

    test('textAlign', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          textAlign="right"
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('isDisabled', () => {
      const { container } = render(
        <OuiCard title="Card title" description="Card description" isDisabled />
      );

      expect(container).toMatchSnapshot();
    });

    describe('paddingSize', () => {
      SIZES.forEach((size) => {
        test(`${size} is rendered`, () => {
          const { container } = render(
            <OuiCard
              title="Card title"
              description="Card description"
              paddingSize={size}
            />
          );

          expect(container).toMatchSnapshot();
        });
      });
    });

    describe('display', () => {
      COLORS.forEach((color) => {
        test(`${color} is rendered`, () => {
          const { container } = render(
            <OuiCard
              title="Card title"
              description="Card description"
              display={color}
            />
          );

          expect(container).toMatchSnapshot();
        });
      });
    });

    test('selectable', () => {
      const { container } = render(
        <OuiCard
          title="Card title"
          description="Card description"
          selectable={{
            onClick: () => {},
          }}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });

  test('horizontal selectable', () => {
    const { container } = render(
      <OuiCard
        title="Card title"
        description="Card description"
        layout="horizontal"
        selectable={{
          onClick: () => {},
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('betaBadgeProps renders href', () => {
    const { container } = render(
      <OuiCard
        title="Card title"
        description="Card description"
        betaBadgeProps={{
          href: 'http://www.elastic.co/',
        }}
        betaBadgeLabel="Link"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
