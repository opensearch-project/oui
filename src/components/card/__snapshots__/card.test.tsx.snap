// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiCard betaBadgeProps renders href 1`] = `
<div
  class="ouiPanel ouiPanel--paddingLarge ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned ouiCard--hasBetaBadge"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
  <span
    class="ouiCard__betaBadgeWrapper"
  >
    <a
      class="ouiBetaBadge ouiBetaBadge-isClickable ouiBetaBadge--hollow ouiCard__betaBadge"
      href="http://www.elastic.co/"
      id="generated-idBetaBadge"
      rel=""
      title="Link"
    >
      Link
    </a>
  </span>
</div>
`;

exports[`OuiCard horizontal selectable 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiPanel--isClickable ouiCard ouiCard--centerAligned ouiCard--horizontal ouiCard--isClickable ouiCard--isSelectable ouiCard--isSelectable--text"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
  <button
    aria-checked="false"
    aria-describedby="generated-idTitle generated-idDescription"
    class="ouiButtonEmpty ouiButtonEmpty--text ouiButtonEmpty--xSmall ouiCardSelect ouiCardSelect--text"
    role="switch"
    type="button"
  >
    <span
      class="ouiButtonContent ouiButtonEmpty__content"
    >
      <span
        class="ouiButtonEmpty__text"
      >
        Select
      </span>
    </span>
  </button>
</div>
`;

exports[`OuiCard is rendered 1`] = `
<div
  aria-label="aria-label"
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props a null icon 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props accepts div props like style 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
  style="min-width:0"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props children 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiCard__children"
    >
      Child
    </div>
  </div>
</div>
`;

exports[`OuiCard props children with description 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
    <div
      class="ouiCard__children"
    >
      Child
    </div>
  </div>
</div>
`;

exports[`OuiCard props display accent is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--accent ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display danger is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--danger ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display plain is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display primary is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--primary ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display subdued is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--subdued ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display success is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--success ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display transparent is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--transparent ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props display warning is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--warning ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props footer 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
  <div
    class="ouiCard__footer"
  >
    <span>
      Footer
    </span>
  </div>
</div>
`;

exports[`OuiCard props hasBorder 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiPanel--hasBorder ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props horizontal 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned ouiCard--horizontal"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props href supports href as a link 1`] = `
<OuiCard
  description="There"
  href="#"
  title="Hoi"
>
  <OuiPanel
    className="ouiCard ouiCard--centerAligned ouiCard--isClickable"
    element="div"
    hasShadow={true}
    onClick={[Function]}
  >
    <div
      className="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiPanel--isClickable ouiCard ouiCard--centerAligned ouiCard--isClickable"
      onClick={[Function]}
    >
      <div
        className="ouiCard__content"
      >
        <OuiTitle
          className="ouiCard__title"
          id="generated-idTitle"
          size="s"
        >
          <span
            className="ouiTitle ouiTitle--small ouiCard__title"
            id="generated-idTitle"
          >
            <a
              aria-describedby="generated-idDescription"
              className="ouiCard__titleAnchor"
              href="#"
              rel="noreferrer"
            >
              Hoi
            </a>
          </span>
        </OuiTitle>
        <OuiText
          className="ouiCard__description"
          id="generated-idDescription"
          size="s"
        >
          <div
            className="ouiText ouiText--small ouiCard__description"
            id="generated-idDescription"
          >
            <p>
              There
            </p>
          </div>
        </OuiText>
      </div>
    </div>
  </OuiPanel>
</OuiCard>
`;

exports[`OuiCard props icon 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned ouiCard--hasIcon"
>
  <div
    class="ouiCard__top"
  >
    <span
      class="myIconClass ouiCard__icon"
      data-ouiicon-type="apmApp"
    />
  </div>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props image 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__top"
  >
    <div
      class="ouiCard__image"
    >
      <div>
        <img
          alt="Nature"
          src="https://source.unsplash.com/400x200/?Nature"
        />
      </div>
    </div>
  </div>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props isDisabled 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--subdued ouiPanel--noShadow ouiPanel--noBorder ouiCard ouiCard--centerAligned ouiCard-isDisabled"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      <button
        aria-describedby=" generated-idDescription"
        class="ouiCard__titleButton"
        disabled=""
      >
        Card title
      </button>
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props paddingSize l is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingLarge ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props paddingSize m is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props paddingSize none is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props paddingSize s is rendered 1`] = `
<div
  class="ouiPanel ouiPanel--paddingSmall ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props selectable 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiPanel--isClickable ouiCard ouiCard--centerAligned ouiCard--isClickable ouiCard--isSelectable ouiCard--isSelectable--text"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
  <button
    aria-checked="false"
    aria-describedby="generated-idTitle generated-idDescription"
    class="ouiButtonEmpty ouiButtonEmpty--text ouiButtonEmpty--xSmall ouiCardSelect ouiCardSelect--text"
    role="switch"
    type="button"
  >
    <span
      class="ouiButtonContent ouiButtonEmpty__content"
    >
      <span
        class="ouiButtonEmpty__text"
      >
        Select
      </span>
    </span>
  </button>
</div>
`;

exports[`OuiCard props textAlign 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--rightAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props titleElement 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <h4
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </h4>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props titleElement with nodes 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <h4
      class="ouiTitle ouiTitle--small ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </h4>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`OuiCard props titleSize 1`] = `
<div
  class="ouiPanel ouiPanel--paddingMedium ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--hasShadow ouiCard ouiCard--centerAligned"
>
  <div
    class="ouiCard__content"
  >
    <span
      class="ouiTitle ouiTitle--xsmall ouiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="ouiText ouiText--small ouiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;
